neuro_san/service/grpc/grpc_agent_service.py

```diff
@@ -89,7 +95,9 @@ def Function(self, request: service_messages.FunctionRequest,
 
         # Get our args in order to pass to grpc-free session level
         request_dict: Dict[str, Any] = MessageToDict(request)
-        response_dict: Dict[str, Any] = self.service.function(request_dict, request_metadata, context)
+        service: AgentService = self.service_provider.get_service()
+        response_dict: Dict[str, Any] =\
+            service.function(request_dict, request_metadata, context)
 
```

Comment:
Here and below - get AgentService from AgentServiceProvider for actual request processing.